### ISSUE TITLE
False positive when combining literals and overloads

### ISSUE THREAD

**Issue by PeterJCLaw** [facebook/pyre-check#258]
Given the following code, `pyre` outputs an error that the first overload signature is not valid for the given implementation, which I believe is incorrect.

``` python
from typing import overload
from typing_extensions import Literal

@overload
def func(*, the_bool: Literal[True]) -> None:  # Incompatible overload [43]:
    # The implementation of `func` does not accept all possible arguments of
    # overload defined on line X
    ...

@overload
def func(the_bool: Literal[False] = False) -> None:
    ...

@overload
def func(the_bool: bool = False) -> None:
    ...

def func(the_bool: bool = False) -> None:
    print(the_bool)
```

My [original use-case](https://github.com/PeterJCLaw/srcomp/blob/9722ee6da5484fc6574b0e759c7769085f86d8eb/sr/comp/raw_compstate.py#L130-L163) for this is that the value of the boolean controls the kind of the return type (which is explicit on the first two overloads and a `Union` on the third and the implementation), but that wasn't needed to reproduce the issue.

I believe that this is a false positive error because the actual implementation will accept `the_bool` as a kwarg, which is what the first overload requires.

**Comment by arthaud** [comment]
Hi @PeterJCLaw,

If you remove the `*` parameter in the first overload, this works properly. We don't handle correctly `*` parameters in overloads, this is a known issue.

**Comment by PeterJCLaw** [comment]
@arthaud thanks for the response.

Unfortunately my [original use-case](https://github.com/PeterJCLaw/srcomp/blob/9722ee6da5484fc6574b0e759c7769085f86d8eb/sr/comp/raw_compstate.py#L130-L163) has other arguments such that removing the `*` changes the overload to no longer be valid for the actual implementation.

If you've got another task for `*` in overloads feel free to close this as a duplicate though.

**Comment by yangdanny97** [comment]
Closing because the reported error no longer repros in the playground

https://pyre-check.org/play?input=from%20typing%20import%20overload%0Afrom%20typing_extensions%20import%20Literal%0A%0A%40overload%0Adef%20func(*%2C%20the_bool%3A%20Literal%5BTrue%5D)%20-%3E%20None%3A%0A%20%20%20%20...%0A%0A%40overload%0Adef%20func(the_bool%3A%20Literal%5BFalse%5D%20%3D%20False)%20-%3E%20None%3A%0A%20%20%20%20...%0A%0A%40overload%0Adef%20func(the_bool%3A%20bool%20%3D%20False)%20-%3E%20None%3A%0A%20%20%20%20...%0A%0Adef%20func(the_bool%3A%20bool%20%3D%20False)%20-%3E%20None%3A%0A%20%20%20%20print(the_bool)
